micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -172,12 +177,15 @@ void checkAndBindMetrics(MeterRegistry registry) {
                 // Kafka has metrics with lower number of tags (e.g. with/without topic or partition tag)
                 // Remove meters with lower number of tags
                 boolean hasLessTags = false;
-                for (Meter other : registry.find(meterName).meters()) {
+                List<Meter> metersForName = registryMetersByNames.getOrDefault(meterName, emptyList());
+                List<Meter> metersWithLessTags = new ArrayList<>(metersForName.size());
+                for (Meter other : metersForName) {
```

Comment:
Is `metersWithLessTags` and `metersForName` really used here?
How about doing this:
```suggestion
                for (Meter other : registryMetersByNames.getOrDefault(meterName, emptyList())) {
```

And also removing `metersWithLessTags.add(other);` and `metersForName.removeAll(metersWithLessTags);`